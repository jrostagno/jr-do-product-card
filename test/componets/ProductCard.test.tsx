import React from 'react';

import { render } from '@testing-library/react';

import { ProductCard } from '../../src/components/ProductCard';
import { product1 } from '../data/product';

describe('ProductTitle', () => {
  test('Debe mostrar el componente ', () => {
    const { asFragment } = render(
      <ProductCard product={product1}>
        {() => <h1>Product card</h1>}
      </ProductCard>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test('Debe incrementar el contador', () => {
    const { asFragment } = render(
      <ProductCard product={product1}>
        {({ count, increaseBy }) => (
          <>
            <h1>Product card</h1>
            <span>{count}</span>
            <button onClick={() => increaseBy(2)}> increase</button>
          </>
        )}
      </ProductCard>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
