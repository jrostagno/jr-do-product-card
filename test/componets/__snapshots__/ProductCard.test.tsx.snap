// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle Debe incrementar el contador 1`] = `
<DocumentFragment>
  <div
    class=" productCard undefined"
  >
    <h1>
      Product card
    </h1>
    <span>
      0
    </span>
    <button>
       increase
    </button>
  </div>
</DocumentFragment>
`;

exports[`ProductTitle Debe mostrar el componente  1`] = `
<DocumentFragment>
  <div
    class=" productCard undefined"
  >
    <h1>
      Product card
    </h1>
  </div>
</DocumentFragment>
`;
